### ISSUE TITLE
抓取报错

### ISSUE THREAD

**Issue by shenyou666** [getmaxun/maxun#126]
reject(new FetchError(`request to ${request.url} failed, reason: ${err.message}`, 'system', err));
[0]           ^
[0] FetchError: request to https://raw.githubusercontent.com/ghostery/adblocker/master/packages/adblocker/assets/easylist/easyprivacy.txt failed, reason: Hostname/IP does not match certificate's altnames: Host: raw.githubusercontent.com. is not in the cert's altnames: DNS:default.ssl.fastly.net, DNS:*.hosts.fastly.net, DNS:*.fastly.com     
[0]     at ClientRequest.<anonymous> (E:\code_project\ftx\maxun\node_modules\node-fetch\lib\index.js:1501:11)

**Comment by tanzhiquan** [comment]
Has it been fixed?

**Comment by tanzhiquan** [comment]
backend-1   | /app/node_modules/node-fetch/lib/index.js:1501
backend-1   | 			reject(new FetchError(`request to ${request.url} failed, reason: ${err.message}`, 'system', err));
backend-1   |           ^
backend-1   | FetchError: request to https://raw.githubusercontent.com/ghostery/adblocker/master/packages/adblocker/assets/peter-lowe/serverlist.txt failed, reason: connect ECONNREFUSED 0.0.0.0:443

**Comment by amhsirak** [comment]
@tanzhiquan @shenyou666 finding it hard to reproduce. As a hot fix, I'll remove this plugin. New release will be in a few hours.

**Comment by jerry-f** [comment]
[0] Node.js v18.20.3
[0] Worker exited with code: 1
[0] Database connected successfully
[0] Database synced successfully!
[0] 2024-11-23T03:52:59.564Z info: Server listening on port 8080
[0] MinIO connected successfully.
[0] 2024-11-23T03:52:59.856Z warn: No active browser in the pool
[0] 2024-11-23T03:52:59.859Z warn: No active browser in the pool
[0] 2024-11-23T03:53:31.840Z warn: No active browser in the pool
[0] 2024-11-23T03:53:31.844Z warn: No active browser in the pool
[0] 2024-11-23T03:53:31.867Z warn: No active browser in the pool
[0] (node:468876) DeprecationWarning: uuidv4() is deprecated. Use v4() from the uuid module instead.
[0] (Use `node --trace-deprecation ...` to show where the warning was created)
[0] 2024-11-23T03:53:32.187Z info: Client connected mmwvmTFnyNDSCVwFAAAB
[0] 2024-11-23T03:53:32.187Z warn: No active browser in the pool
[0] /home/fs/pro/pro-list/maxun/node_modules/cross-fetch/node_modules/node-fetch/lib/index.js:1501
[0]                     reject(new FetchError(`request to ${request.url} failed, reason: ${err.message}`, 'system', err));
[0]           ^
[0] FetchError: request to https://raw.githubusercontent.com/ghostery/adblocker/master/packages/adblocker/assets/easylist/easylist.txt failed, reason: connect ECONNREFUSED 0.0.0.0:443
[0]     at ClientRequest.<anonymous> (/home/fs/pro/pro-list/maxun/node_modules/cross-fetch/node_modules/node-fetch/lib/index.js:1501:11)
[0]     at ClientRequest.emit (node:events:517:28)
[0]     at ClientRequest.emit (node:domain:489:12)
[0]     at TLSSocket.socketErrorListener (node:_http_client:501:9)
[0]     at TLSSocket.emit (node:events:517:28)
[0]     at TLSSocket.emit (node:domain:489:12)
[0]     at emitErrorNT (node:internal/streams/destroy:151:8)
[0]     at emitErrorCloseNT (node:internal/streams/destroy:116:3)
[0]     at processTicksAndRejections (node:internal/process/task_queues:82:21) {
[0]   type: 'system',
[0]   errno: 'ECONNREFUSED',
[0]   code: 'ECONNREFUSED'
[0] }
[0] [nodemon] app crashed - waiting for file changes before starting...

-----------------------------

[Start Training Robot ] error info

**Comment by jerry-f** [comment]
> [0] Node.js v18.20.3[0] Worker 退出，代码为 1[0] 已成功连接数据库[0] 数据库已成功同步！[0] 2024-11-23T03：52：59.564Z 信息：服务器侦听端口 8080[0] MinIO 连接成功。[0] 2024-11-23T03：52：59.856Z 警告：池中没有活动的浏览器[0] 2024-11-23T03：52：59.859Z 警告：池中没有活动的浏览器[0] 2024-11-23T03：53：31.840Z 警告：池中没有活动的浏览器[0] 2024-11-23T03：53：31.844Z 警告：池中没有活动的浏览器[0] 2024-11-23T03：53：31.867Z 警告：池中没有活动浏览器[0] （node：468876） DeprecationWarning： uuidv4（） 已弃用。请改用 uuid 模块中的 v4（）。[0]（使用`节点 --trace-deprecation ...` 显示警告的创建位置）[0] 2024-11-23T03：53：32.187Z 信息：客户端已连接 mmwvmTFnyNDSCVwFAAAB[0] 2024-11-23T03：53：32.187Z 警告：池中没有活动的浏览器[0] /首页/fs/pro/pro-list/maxun/node_modules/cross-fetch/node_modules/node-fetch/lib/index.js：1501[0] 拒绝（new FetchError（ `request to ${request.url} failed, reason: ${err.message}` ， 'system'， err））; [0] ^[0] FetchError：请求 https://raw.githubusercontent.com/ghostery/adblocker/master/packages/adblocker/assets/easylist/easylist.txt 失败，原因：连接 ECONNREFUSED 0.0.0.0：443[0] 在 ClientRequest 上。（/home/fs/pro/pro-list/maxun/node_modules/cross-fetch/node_modules/node-fetch/lib/index.js：1501：11）[0] 在 ClientRequest.emit （node：events：517：28）[0] 在 ClientRequest.emit （node：domain：489：12）[0] 在 TLSSocket.socketErrorListener （node：_http_client：501：9）[0] 在 TLSSocket.emit （node：events：517：28）[0] 在 TLSSocket.emit （node：domain：489：12）[0] 在 emitErrorNT （node：internal/streams/destroy：151：8）[0] 在 emitErrorCloseNT （node：internal/streams/destroy：116：3）[0] 在 processTicksAndRejections （node：internal/process/task_queues：82：21） {[0] 类型： 'system'，[0] errno： 'ECONNREFUSED'，[0] 代码： 'ECONNREFUSED' [0] }[0] [nodemon] 应用程序崩溃 - 在启动之前等待文件更改...
> 
> [启动训练机器人] 错误信息

version is [v0.0.2](https://github.com/getmaxun/maxun/releases/tag/v0.0.2)

**Comment by amhsirak** [comment]
@jerry-f @tanzhiquan @shenyou666 are any of you behind a firewall?

I am finding it hard to reproduce this.

If you can share more about your setup it would be very helpful

**Comment by Ycituss** [comment]
@amhsirak I ran into the same problem, I think it may be that we are in China, blocked by the Great firewall, and do not have direct access to GitHub

**Comment by Ycituss** [comment]
I found a solution. I changed my host file, allows me to visit raw.githubusercontent.com via IP directly rather than through a DNS lookup, problem is solved

**Comment by Neymarliqi** [comment]
I've already solved the problem:
1、The core reason is that it's impossible to access the address https://raw.githubusercontent.com/ghostery/adblocker/master/packages/adblocker/assets/peter-lowe/serverlist.txt stably in China.
2、This is caused by the fact that there is this dependency in package.json.[pic]
![e22b95139e6d8179e38485ad45a4345](https://github.com/user-attachments/assets/fab84095-b301-4835-b188-fc1a9a967659)
3、The solution is that I used the mirror address https://ghproxy.com/ of https://raw.githubusercontent.com/, with the IP: 129.146.41.82, and configured it in the hosts file on my computer, just like this: 129.146.41.82 raw.githubusercontent.com. In this way, I can access it successfully.

**Comment by ruanjunmin** [comment]
> I've already solved the problem: 1、The core reason is that it's impossible to access the address https://raw.githubusercontent.com/ghostery/adblocker/master/packages/adblocker/assets/peter-lowe/serverlist.txt stably in China. 2、This is caused by the fact that there is this dependency in package.json.[pic] ![e22b95139e6d8179e38485ad45a4345](https://private-user-images.githubusercontent.com/21285451/391980202-fab84095-b301-4835-b188-fc1a9a967659.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM4ODE3MjEsIm5iZiI6MTczMzg4MTQyMSwicGF0aCI6Ii8yMTI4NTQ1MS8zOTE5ODAyMDItZmFiODQwOTUtYjMwMS00ODM1LWIxODgtZmMxYTlhOTY3NjU5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjExVDAxNDM0MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNmMjQ4ZTg2ZDhiNTAyNjk2YjgxODE0MDQxNTZiNDFkMjQxNWVmZjk5ZWIxNTY4OTAyOGZlZThiMTZjNmYxZGUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ib6p7hJlUnS7fVw_sg8SA-8BsbFxkifCPt-n2fN1teM) 3、The solution is that I used the mirror address https://ghproxy.com/ of https://raw.githubusercontent.com/, with the IP: 129.146.41.82, and configured it in the hosts file on my computer, just like this: 129.146.41.82 raw.githubusercontent.com. In this way, I can access it successfully.

bro, can you tell me how to do that? 大哥怎么改教下

**Comment by ruanjunmin** [comment]
> > I've already solved the problem: 1、The core reason is that it's impossible to access the address https://raw.githubusercontent.com/ghostery/adblocker/master/packages/adblocker/assets/peter-lowe/serverlist.txt stably in China. 2、This is caused by the fact that there is this dependency in package.json.[pic] ![e22b95139e6d8179e38485ad45a4345](https://private-user-images.githubusercontent.com/21285451/391980202-fab84095-b301-4835-b188-fc1a9a967659.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM4ODE3MjEsIm5iZiI6MTczMzg4MTQyMSwicGF0aCI6Ii8yMTI4NTQ1MS8zOTE5ODAyMDItZmFiODQwOTUtYjMwMS00ODM1LWIxODgtZmMxYTlhOTY3NjU5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjExVDAxNDM0MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNmMjQ4ZTg2ZDhiNTAyNjk2YjgxODE0MDQxNTZiNDFkMjQxNWVmZjk5ZWIxNTY4OTAyOGZlZThiMTZjNmYxZGUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ib6p7hJlUnS7fVw_sg8SA-8BsbFxkifCPt-n2fN1teM) 3、The solution is that I used the mirror address https://ghproxy.com/ of https://raw.githubusercontent.com/, with the IP: 129.146.41.82, and configured it in the hosts file on my computer, just like this: 129.146.41.82 raw.githubusercontent.com. In this way, I can access it successfully.
> 
> bro, can you tell me how to do that? 大哥怎么改教下

![image](https://github.com/user-attachments/assets/b24ba535-e225-4aa4-ae54-e39ec28ebfcf)
will it work?

**Comment by amhsirak** [comment]
@ruanjunmin can you please try with the latest  version? I've changed the way we access the adblocking list. Do let me know if it works or not.

**Comment by ruanjunmin** [comment]
> > @ruanjunmin can you please try with the latest version? I've changed the way we access the adblocking list. Do let me know if it works or not.
> 
> Okey, Im gonna try the latest version out

The maxun backend still cannot connect correctly, and the error message is as follows:

**Comment by ruanjunmin** [comment]
Also, after clicking the [create robot] button and entering the URL, the error message is as follows:
![image](https://github.com/user-attachments/assets/d76441fc-e8bf-4118-ab81-2a09df25c054)

F:\maxun>docker-compose logs -f backend
time="2024-12-11T22:39:16+08:00" level=warning msg="F:\\maxun\\docker-compose.yml: the attribute `version` is obsolete, it will be ignored, please remove it to avoid potential confusion"
backend-1  |
backend-1  | > maxun@0.0.4 server
backend-1  | > cross-env NODE_OPTIONS='--max-old-space-size=8000' nodemon server/src/server.ts
backend-1  |
backend-1  | [nodemon] 2.0.22
backend-1  | [nodemon] to restart at any time, enter `rs`
backend-1  | [nodemon] watching path(s): *.*
backend-1  | [nodemon] watching extensions: ts,json
backend-1  | [nodemon] starting `ts-node server/src/server.ts`
backend-1  | Worker is running...
backend-1  | Connected to Redis!
backend-1  | MinIO connected successfully.
backend-1  | Database connected successfully
backend-1  | Database synced successfully!
backend-1  | 2024-12-11T14:37:49.442Z info: Server listening on port 8080
backend-1  | 2024-12-11T14:38:18.668Z warn: No active browser in the pool
backend-1  | 2024-12-11T14:42:35.212Z warn: No active browser in the pool
backend-1  | 2024-12-11T14:42:35.243Z warn: No active browser in the pool
backend-1  | (node:39) DeprecationWarning: uuidv4() is deprecated. Use v4() from the uuid module instead.
backend-1  | (Use `node --trace-deprecation ...` to show where the warning was created)
backend-1  | 2024-12-11T14:42:35.303Z info: Client connected urjxlK1rlKedOD_GAAAB
backend-1  | 2024-12-11T14:42:35.304Z warn: No active browser in the pool
backend-1  | /app/node_modules/node-fetch/lib/index.js:1501
backend-1  |                    reject(new FetchError(`request to ${request.url} failed, reason: ${err.message}`, 'system', err));
backend-1  |           ^
backend-1  | FetchError: request to https://raw.githubusercontent.com/ghostery/adblocker/master/packages/adblocker/assets/ublock-origin/resources.json failed, reason: connect ECONNREFUSED 0.0.0.0:443
backend-1  |     at ClientRequest.<anonymous> (/app/node_modules/node-fetch/lib/index.js:1501:11)
backend-1  |     at ClientRequest.emit (node:events:519:28)
backend-1  |     at ClientRequest.emit (node:domain:488:12)
backend-1  |     at TLSSocket.socketErrorListener (node:_http_client:500:9)
backend-1  |     at TLSSocket.emit (node:events:519:28)
backend-1  |     at TLSSocket.emit (node:domain:488:12)
backend-1  |     at emitErrorNT (node:internal/streams/destroy:169:8)
backend-1  |     at emitErrorCloseNT (node:internal/streams/destroy:128:3)
backend-1  |     at processTicksAndRejections (node:internal/process/task_queues:82:21) {
backend-1  |   type: 'system',
backend-1  |   errno: 'ECONNREFUSED',
backend-1  |   code: 'ECONNREFUSED'
backend-1  | }
backend-1  | [nodemon] app crashed - waiting for file changes before starting...

**Comment by amhsirak** [comment]
@ruanjunmin Thanks for the reporting the new issue. Turns out the old issue is solved but under the hood, the plugin is making a request to another URL which cannot be accessed in China. Will look into the plugin to figure out how to disable this :)

**Comment by ruanjunmin** [comment]
> @ruanjunmin感谢您报告新问题。原来旧问题已经解决，但实际上，该插件正在向另一个在中国无法访问的 URL 发出请求。我会研究该插件以找出如何禁用它 :)

thanks,bro. I hope this bug can work out ASAP, i can't wait to use Maxun~

**Comment by amhsirak** [comment]
Closing as v0.0.5 fixed this. If anyone faces this issue after upgrading - feel free to reopen
